src/data/mv_polynomial.lean

```diff
@@ -2,12 +2,81 @@
 Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Johan Commelin, Mario Carneiro
-
-Multivariate Polynomial
 -/
 import algebra.ring
 import data.finsupp data.polynomial data.equiv.algebra
 
+/-!
+# Multivariate polynomials
+
+This file defines polynomial rings over a base ring (or even semiring),
+with variables from a general type `σ` (which could be infinite).
+
+## Important definitions
+
+Let `R` be a commutative ring (or a semiring) and let `σ` be an arbitrary
+type. This file creates the type `mv_polynomial σ R`, which mathematicians
+might denote `R[X_i : i ∈ σ]`. It is the type of multivariate
+(a.k.a. multivariable) polynomials, with variables
+corresponding to the terms in `σ`, and coefficients in `R`.
+
+In the definitions below, we use the following notation:
+`σ : Type*` (indexing the variables)
```

Comment:
The following list will not be typeset as a list since there is no markdown. I noticed this by looking at https://observablehq.com/@bryangingechen/github-lean-doc-preview?url=%221892%22 which is extremely handy especially for doc PRs.

Reply:
!! thank you so much for this link and thank you Bryan for setting this up!